File: youlexuan_parent/youlexuan_manager_web/src/main/java/com/offcn/sellergoods/controller/BrandController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.offcn.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController<tbbrands> {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page, int size) {
        return brandService.findPage(page, size);
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand, int page, int size) {
        return  brandService.findPage(brand,page,size);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return  new Result(true, "新增品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
            return  new Result(false, "新增品牌失败");
        }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            return  new Result(true, "修改品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
            return  new Result(false, "修改品牌失败");
        }
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            brandService.delete(ids);
            return  new Result(true, "删除品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
            return  new Result(false, "删除品牌失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }


    @RequestMapping("/findBrandOptionList")
    public List<Map> findBrandOptionList() {
        return brandService.findBrandOptionList();
    }

}

File: youlexuan_parent/youlexuan_manager_web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function($scope, $controller, brandService){
    // 继承
    $controller('baseController', {
        $scope: $scope
    });

    $scope.findAll = function() {
        brandService.findAll().success(function (resp) {
            $scope.list = resp;
        })
    }

    $scope.findPage = function(page, size) {
        brandService.findPage(page, size).success(function(resp){
            $scope.list = resp.rows;
            $scope.paginationConf.totalItems = resp.total;
        })
    }

    // 保存、修改
    $scope.save = function() {
        brandService.save($scope.entity).success(function(resp){
            if(resp.success) {
                $scope.reloadList();
            } else {
                alert(resp.message);
            }
        });
    }

    // 查询单个品牌
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (resp) {
            if(resp) {
                $scope.entity = resp;
            }
        })
    }

    // 删除
    $scope.delete = function () {
        if($scope.selectIds.length > 0) {
            brandService.delete($scope.selectIds).success(function (resp) {
                if(resp.success) {
                    $scope.reloadList();
                    $scope.selectIds = [];
                } else {
                    alert(resp.message);
                }
            })
        } else {
            alert("请选中要删除的记录！");
        }

    }

    $scope.searchEntity = {};
    // 查询
    $scope.search = function (page, size) {
        brandService.search(page, size, $scope.searchEntity).success(function (resp) {
            $scope.list = resp.rows;
            $scope.paginationConf.totalItems = resp.total;
        });
    }

});
